hardware/raspberrypi/bcm2711_bootloader_config.md

```diff
@@ -225,3 +274,43 @@ rpi-eeprom-config --out pieeprom-netboot.bin --config bootconf.txt pieeprom.bin
 sudo rpi-eeprom-update -d -f ./pieeprom-netboot.bin
 sudo reboot
 ```
+
+## USB mass storage boot
+This is only available in the BETA release and requires updated (not yet released) firmware via rpi-update. If you aren't already familiar with how to use a USB drive for the root filesystem then you probably want to wait until this feature is in the default Raspbian image.
+
+### USB_MSD_EXCLUDE_VID_PID
+A list of up to 4 VID/PID pairs specifying devices which the bootloader should ignore. If this matches a HUB then the HUB won’t be enumerated causing all downstream devices to be excluded.
+This is intended to allow problematic (e.g. very slow to enumerate) devices to be ignored during boot enumeration. This is specific to the bootloader and is not passed to the OS.
+
+The format is a comma separated list of hexadecimal values with the VID as most significant nibble. Spaces are not allowed.
+E.g. 0347000a0,0a4231234
+
+Default: “”  
+Version: pieeprom-2020-05-15.bin - BETA    
+
+### USB_MSD_DISCOVER_TIMEOUT
+If no USB mass storage devices are found within this timeout then USB-MSD is stopped and the next boot mode is selected
+
+Default: 20000 (20 seconds)  
+Version: pieeprom-2020-05-15.bin - BETA  
+
+### USB_MSD_LUN_TIMEOUT
+How long to wait in milliseconds before advancing to the next LUN e.g. a multi-slot SD-CARD reader. This is still being tweaked but may help speed up boot if old/slow devices are connected as well as fast USB-MSD device containing the OS.
+
+Default:  2000 (2 seconds)  
+Version: pieeprom-2020-05-15.bin - BETA  
+
+### XHCI_DEBUG
+A bit field Controls the verbosity of USB trace message for mass storage boot mode. Enabling all of these messages generates a huge amount of log data which will slow down booting and may even cause boot to fail. For verbose logs it's best to use NET_CONSOLE
+
+* Bit 0 - USB descriptors
+* Bit 1 - Mass storage mode state machine 
+* Bit 2 - Mass storage mode state machine - verbose
+* Bit 3 - All USB requests
+* Bit 4 - Log device and hub state machines
+* Bit 5 - Log all xHCI TRBs (VERY VERBOSE)
+* Bit 6 - Logg all xHCI events (VERY VERBOSE)
+
+Default: 0
+Version: pieeprom-2020-05-15.bin - BETA  
+
```

Comment:
Default 0 means no logging by default. How do you specify the bits - is it hex like BOOT_ORDER? Might be worth adding an example of how to set a non-default value to explain this.

Reply:
added example